codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -43,12 +43,24 @@ public final DnsQuestion decodeQuestion(ByteBuf in) throws Exception {
     @Override
     public final <T extends DnsRecord> T decodeRecord(ByteBuf in) throws Exception {
         final String name = decodeName(in);
+
+        final int endOffset = in.writerIndex();
+        if (endOffset - in.readerIndex() < 10) {
+            // Not enough data
+            return null;
+        }
+
         final DnsRecordType type = DnsRecordType.valueOf(in.readUnsignedShort());
         final int aClass = in.readUnsignedShort();
         final long ttl = in.readUnsignedInt();
         final int length = in.readUnsignedShort();
         final int offset = in.readerIndex();
 
+        if (endOffset - offset < length) {
```

Comment:
We can do that, although we are not using that behavior because we simply stop decoding.